collect_app/src/main/java/org/odk/collect/android/utilities/FormDownloadBroadcastHelper.java

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2018 Ephraim Kigamba
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License
+*/
+
+package org.odk.collect.android.utilities;
+
+import android.content.Context;
+import android.content.Intent;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+/**
+ * This helper broadcasts the result of the form download initiated in {@link org.odk.collect.android.services.FormDownloadService}
+ *
+ * @author Ephraim Kigamba (nek.eam@gmail.com)
+ */
+
+public final class FormDownloadBroadcastHelper {
```

Comment:
Hey @ekigamba! What do you think about putting the contents of this helper inside `FormDownloadService`? Especially since it's just a broadcast from the service itself.

Reply:
@jd-alexander It's also shared with the FormDownloadJob

Reply:
How about doing all the error checking in the `FormDownloadJob` so the method can live there? Seems you can pass on a `null` or empty `formID` and do the checking for those cases there?

Reply:
Okay no prob!

Reply:
@lognaturel that's an idea